scripts/cmake/Utils.cmake

```diff
@@ -48,9 +49,14 @@ function(mangle output_c dir)
     set(single_file_desc "${MANGLE_WILDCARD} in ${dir}/")
     file(GLOB inputs RELATIVE ${dir} CONFIGURE_DEPENDS ${dir}/${MANGLE_WILDCARD} ${MANGLE_UNPARSED_ARGUMENTS})
   else()
-    set(inputs ${MANGLE_UNPARSED_ARGUMENTS})
-    list(JOIN ", " inputs single_file_desc)
-    set(single_file_desc "${inputs} in ${dir}")
+    if(MANGLE_RECURSIVE)
+      set(single_file_desc "${MANGLE_WILDCARD} in ${dir}/")
+      file(GLOB_RECURSE inputs RELATIVE ${dir} CONFIGURE_DEPENDS ${MANGLE_UNPARSED_ARGUMENTS} "${dir}/*")
```

Comment:
In this branch, MANGE_WILDCARD is not set. If you keep this functionality here, I recommend to remove it from the `single_file_desc` and also to update the comment on the WILDCARD keyword parameter above to reflect what's going on. It looks a bit inconsistent to have globbing when no WILDCARD parameter is specified.

Reply:
Hm, I will invert it (check for MANGLE_RECURSIVE first) and also use the wildcard there.

Reply:
Many thanks!